src/scripts/main.js

```diff
@@ -2,46 +2,42 @@
 
 const spider = document.querySelector('.spider');
 const theWall = document.querySelector('.wall');
-const theWallWidth = parseInt(getComputedStyle(theWall).width);
-const theWallHeight = parseInt(getComputedStyle(theWall).height);
-const theWallBorder = parseInt(getComputedStyle(theWall).border);
+const theWallWidth = parseInt(theWall.clientWidth);
+const theWallHeight = parseInt(theWall.clientHeight);
+const theWallBorder = parseInt(theWall.clientTop);
 
 document.addEventListener('click', e => {
-  if (e.target.closest('.wall') || e.target.className === 'spider') {
-    if (e.clientX <= theWall.offsetLeft + theWallBorder + spider.width / 2) {
-      spider.style.left = `0px`;
-    } else if (
-      e.clientX >= theWall.offsetLeft
-      + theWallBorder
-      + theWallWidth
-      - spider.width / 2
-    ) {
-      spider.style.left = `${theWallWidth - spider.width}px`;
-    } else {
-      spider.style.left = `${
-        e.clientX
-        - theWall.offsetLeft
-        - theWallBorder
-        - spider.width / 2
-      }px`;
-    }
-
-    if (e.clientY <= theWall.offsetTop + theWallBorder + spider.width / 2) {
-      spider.style.top = `0px`;
-    } else if (
-      e.clientY >= theWall.offsetTop
-      + theWallBorder
-      + theWallHeight
-      - spider.width / 2
-    ) {
-      spider.style.top = `${theWallHeight - spider.width}px`;
-    } else {
-      spider.style.top = `${
-        e.clientY
-        - theWall.offsetTop
-        - theWallBorder
-        - spider.height / 2
-      }px`;
-    }
+  if (!e.target.closest('.wall') || !e.target.className === 'spider') {
+    return;
   }
+
+  spider.style.left = coord(
+    e.clientX, spider.width, theWall.offsetLeft, theWallWidth
+  );
+
+  spider.style.top = coord(
+    e.clientY, spider.height, theWall.offsetTop, theWallHeight
+  );
 });
+
+function coord(mouseCoord, spiderSize, offset, theWallSize) {
+  if (
+    mouseCoord <= offset + theWallBorder + spiderSize / 2
+  ) {
+    return `0px`;
+  } else if (
+    mouseCoord >= offset
+    + theWallBorder
+    + theWallSize
+    - spiderSize / 2
+  ) {
```

Comment:
do you really need such long conditions?
maybe it's better to simplify it further or assign these calculations to the vars, cause now every time js checks these conditions it calculates it again and again